stdlib/sql/to_test.go

```diff
@@ -0,0 +1,281 @@
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/querytest"
+	"github.com/influxdata/flux/stdlib/influxdata/influxdb"
+	fsql "github.com/influxdata/flux/stdlib/sql"
+	"github.com/influxdata/flux/values"
+)
+
+func TestSqlTo(t *testing.T) {
+	tests := []querytest.NewQueryTestCase{
+		{
+			Name: "from with database",
+			Raw:  `import "sql" from(bucket: "mybucket") |> sql.to(driverName:"sqlmock", dataSourceName:"root@/db", table:"TestTable")`,
+			Want: &flux.Spec{
+				Operations: []*flux.Operation{
+					{
+						ID: "from0",
+						Spec: &influxdb.FromOpSpec{
+							Bucket: "mybucket",
+						},
+					},
+					{
+						ID: "toSQL1",
+						Spec: &fsql.ToSQLOpSpec{
+							DriverName:     "sqlmock",
+							DataSourceName: "root@/db",
+							Table:          "TestTable",
+						},
+					},
+				},
+				Edges: []flux.Edge{
+					{Parent: "from0", Child: "toSQL1"},
+				},
+			},
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			querytest.NewQueryTestHelper(t, tc)
+		})
+	}
+}
+
+func TestToSQL_Process(t *testing.T) {
+	driverName := "sqlmock"
+	dsn := "root@/db"
+	_, _, _ = sqlmock.NewWithDSN(dsn)
+	w := &fsql.Writer{
+		A: []string{},
+		B: make([][]interface{}, 0),
+	}
+	fsql.TestingWriter = w
+	type wanted struct {
+		Table  []*executetest.Table
+		Result *fsql.Writer
+	}
+	testCases := []struct {
+		name string
+		spec *fsql.ToSQLProcedureSpec
+		data []flux.Table
+		want wanted
+	}{
+		{
+			name: "coltable with name in _measurement",
+			spec: &fsql.ToSQLProcedureSpec{
+				Spec: &fsql.ToSQLOpSpec{
+					DriverName:     driverName,
+					DataSourceName: dsn,
+					Table:          "TestTable2",
+				},
+			},
+			data: []flux.Table{executetest.MustCopyTable(&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_measurement", Type: flux.TString},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "fred", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(11), "a", 2.0, "one"},
+					{execute.Time(21), "a", 2.0, "one"},
+					{execute.Time(21), "b", 1.0, "seven"},
+					{execute.Time(31), "a", 3.0, "nine"},
+					{execute.Time(41), "c", 4.0, "elevendyone"},
+				},
+			})},
+			want: wanted{
+				Table: []*executetest.Table{{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_measurement", Type: flux.TString},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "fred", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{execute.Time(11), "a", 2.0, "one"},
+						{execute.Time(21), "a", 2.0, "one"},
+						{execute.Time(21), "b", 1.0, "seven"},
+						{execute.Time(31), "a", 3.0, "nine"},
+						{execute.Time(41), "c", 4.0, "elevendyone"},
+					},
+				}},
+				Result: &fsql.Writer{
```

Comment:
I would prefer the process method and the insert statements to be tested in isolation. Is that possible? Maybe it's too difficult?